### PR TITLE
[FEATURE] Add OpenMSDatabasesInfo Util to list supported Enzymes and Mods in tsv

### PR DESCRIPTION
**Pull request by jpfeuffer**
# Description

See title. I think testing would be superfluous and tedious since we would need to adapt the test files every time the DB is updated.

### CODE REVIEW

**Review by Leon-Bichmann — APPROVED** [review]
LGTM, as far as I can see.

File: src/openms/include/OpenMS/CHEMISTRY/ModificationsDB.h [hunk 1]
```diff
@@ -209,7 +209,9 @@ namespace OpenMS
     /// Collects all modifications that can be used for identification searches
     void getAllSearchModifications(std::vector<String>& modifications) const;
 
-protected:
+    void writeCSV(const String& filename);
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/CHEMISTRY/ModificationsDB.h:212]: TSV oder CSV? Please add documentation here what fields are tab separated written.

File: src/openms/source/CHEMISTRY/ProteaseDB.cpp [hunk 2]
```diff
@@ -105,4 +105,13 @@ namespace OpenMS
     }
   }
 
+  void ProteaseDB::writeCSV(String const& filename)
+  {
+    std::ofstream ofs(filename, std::ofstream::out);
+    ofs << "OpenMS_AllowedEnzymes" << "\n";
```
Comment (timosachsenberg) [inline src/openms/source/CHEMISTRY/ProteaseDB.cpp:111]: Just "Enzyme"?

File: src/utils/OpenMSDatabasesInfo.cpp [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  {
+  }
+
+protected:
+  // this function will be used to register the tool parameters
+  // it gets automatically called on tool execution
+  void registerOptionsAndFlags_() override
+  {
+    // Output CSV file
+    registerOutputFile_("enzymes_out", "<out>", "", "Enzymes as CSV", true, false);
```
Comment (timosachsenberg) [inline src/utils/OpenMSDatabasesInfo.cpp:78]: says CSV type is TSV

**Review by timosachsenberg — APPROVED** [review]
Looks good. Only minor comments.
Some brainstorming: could this tool be also be used to conveniently export a json file with all information? Would writing it on std::cout also be an option? Could open a few other possibilities but for what you want to do now it is fine.

File: src/openms/source/CHEMISTRY/ProteaseDB.cpp (hunk 2 above)
Comment (jpfeuffer) [reply src/openms/source/CHEMISTRY/ProteaseDB.cpp:111]: Want to make a clear distinction between header and options.

**Comment by jpfeuffer** [comment]
Stdout is an option later but output files *have to be present* for some workflow engines.
Sure, if someone needs json, this can be done in the same tool.

File: src/openms/include/OpenMS/CHEMISTRY/ModificationsDB.h [hunk 4]
```diff
@@ -209,7 +209,10 @@ namespace OpenMS
     /// Collects all modifications that can be used for identification searches
     void getAllSearchModifications(std::vector<String>& modifications) const;
 
-protected:
+    /// Writes the contents to a TSV file
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/CHEMISTRY/ModificationsDB.h:212]:
```suggestion
    /// Writes tab separated entrie: FullId,FullName,Origin,AA,TerminusSpecificity,DiffMonoMass (including header) to TSV file
```

**Pull request merged**